### PR TITLE
Added rubocop configs

### PR DESCRIPTION
**Pull request by BrandonMathis**
## What changed
- run rubocop before tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,40 +4,15 @@ rvm:
...
-  - printf 'y\ny\ny\ny\ny\nn' | rails new api_admin_app -m smashing-template/template.rb -B
-  - cd api_admin_app
-  - cp config/database.example.yml config/database.yml
-  - cp config/secrets.example.yml config/secrets.yml
-  - bundle exec rake db:migrate --trace
-  - bundle exec rake db:test:prepare --trace
-  - cp ../smashing-template/test/example_spec.rb spec/example_spec.rb
-  - cp ../smashing-template/test/example.feature features/example.feature
-  - cp ../smashing-template/test/example_steps.rb features/example_steps.rb
+  - bin/gen_interactive_app
```
Comment (Ru-T) [inline .travis.yml]: This is awesome, thank you!

File: lib/gem_configurator.rb [hunk 2]
```diff
@@ -46,6 +47,20 @@ def code_climate_config
   end
 end
 
+def rubocop_config
+  inside 'spec' do
+    inject_into_file 'spec_helper.rb', after: "RSpec.configure do |config|\n" do
+        <<-RUBY
+    config.before(:suite) do
```
Comment (Ru-T) [inline lib/gem_configurator.rb]: Lines 54 through 58 are indented one tab too far, causing rubocop to fail in travis on a generated app.

**Comment by Ru-T** [comment]
Thanks @BrandonMathis - the changes to the Travis file are amazing. 

For the spec_helper config: this is interesting - I’m not sure everyone will want this behavior. I often check rubocop at the end of a branch after all tests are passing, rather than before I run my tests. Do you think this should be an enforced best practice?

Defer to your judgment!

**Comment by BrandonMathis** [comment]
I've changed the rubocop config to only run AFTER the test suite on a full successful pass. I find this far more helpful :) Thx for the tip @Ru-T!

**Comment by Ru-T** [comment]
Thanks @BrandonMathis - that sounds great!

**Pull request merged**